### PR TITLE
Person detection selector

### PR DESCRIPTION
**Pull request by fredrikhaggbom**
Adds a select for toggling person detection on/off and setting sensitivity.

Requires [this branch of pytapo](https://github.com/JurajNyiri/pytapo/tree/dev)

Unfortunately the camera does not seem to trigger an onvif event when person detection is triggered, which makes it impossible to add a sensor for person detection as well.

### CODE REVIEW

File: custom_components/tapo_control/camera.py [hunk 1]
```diff
@@ -170,6 +175,12 @@ async def async_update(self) -> None:
             self._attr_motion_detection_enabled = (
                 "enabled" in data and data["enabled"] == "on"
             )
+            data = await self._hass.async_add_executor_job(
+                self._controller.getPersonDetection
+            )
+            self._attr_person_detection_enabled = (
+                "enabled" in data and data["enabled"] == "on"
+            )
```
[the remark this answers is not in the document]
Comment (JurajNyiri) [reply custom_components/tapo_control/camera.py:183]: Also no need to do it in brand property.

File: custom_components/tapo_control/camera.py [hunk 2]
```diff
@@ -202,6 +214,18 @@ async def async_disable_motion_detection(self):
...
+            self._controller.setPersonDetection, True
+        )
+        await self._coordinator.async_request_refresh()
+
+    async def async_disable_person_detection(self):
+        await self.hass.async_add_executor_job(
+            self._controller.setPersonDetection, False
+        )
+        await self._coordinator.async_request_refresh()
+
```
Comment (JurajNyiri) [inline custom_components/tapo_control/camera.py:228]: We have select, no need to add camera functions.

File: custom_components/tapo_control/camera.py [hunk 3]
```diff
@@ -125,6 +126,10 @@ def device_info(self) -> DeviceInfo:
     def motion_detection_enabled(self):
         return self._motion_detection_enabled
 
+    @property
+    def person_detection_enabled(self):
+        return self._person_detection_enabled
+
```
Comment (JurajNyiri) [inline custom_components/tapo_control/camera.py:132]: We have select, no need to add camera attribute.

**Review by JurajNyiri — CHANGES_REQUESTED** [review]
Thank you for your contribution! We do not need to modify camera.py at all for this PR. The rest looks good, I still have to take a look into pytapo changes and test this.

**Comment by JurajNyiri** [comment]
Dev branch of pytapo has been released as 2.9 version having all these functions.

File: custom_components/tapo_control/camera.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (fredrikhaggbom) [reply custom_components/tapo_control/camera.py]: Camera.py has been reset, and no changes are no longer made to the file in this PR

**Review by JurajNyiri — APPROVED** [review]

**Review by JurajNyiri — CHANGES_REQUESTED** [review]
Great work, can we also add `getPersonDetection` into `pytapoFunctionMap` so that the call does not need to be made inside `isCacheSupported` and `check_and_create` reducing the amount of network calls?

**Pull request merged**